test/fetchOptions.test.ts

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2022, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import sinon from "sinon";
+import { BaseClient } from "../src/base/client";
+import { assert } from "chai";
+import proxyquire from "proxyquire";
+
+describe("Fetch Options", () => {
+  const uri = "https://localhost:3000";
+  const fetchStub = sinon.stub();
+  const staticClient = proxyquire("../src/base/staticClient", {
+    "make-fetch-happen": fetchStub,
+  });
+
+  beforeEach(() => {
+    fetchStub.reset();
```

Comment:
Why do you have to reset the stub before every test?

Reply:
From my understanding, stubs maintain a history of calls and other things. I reset the stub before each test to ensure previous calls to the stub don't affect future calls to the stub. It basically boils down to ensuring I have a clean slate for each test.